### ISSUE TITLE
[Support]: White Screen using Firefox when in private browsing mode

### ISSUE THREAD

**Issue by twistedlincoln** [blakeblackshear/frigate#7524]
### Describe the problem you are having

When connecting to Frigate using Firefox (or its derivatives), a blank white screen is the only thing displayed whenever one is in "private browsing mode" -- either by opening a "new private window" or by setting the browser's history setting to "never remember history" within preferences.  Things work fine when in a normal browsing window as long as the preference is set to remember history.

This behavior is not present when connecting from Chrome (or its derivatives).  Frigate loads and works fine regardless of if the browser is in "incognito mode" or is set not to remember history.

Ideally, this issue would be resolved such that one can connect using Firefox while still browsing privately.  However if that is not possible for some reason, it would be preferable to show the user an error message explaining the issue, as this took me several hours of troubleshooting to figure out.

### Version

0.12.1-367D724

### Frigate config file

```yaml
N/A
```

### Relevant log output

```shell
N/A
```

### FFprobe output from your camera

```shell
N/A
```

### Operating system

Other Linux

### Install method

Docker Compose

### Coral version

USB

### Network connection

Wired

### Camera make and model

N/A

**Comment by NickM-27** [comment]
What error do you see in Firefox with this mode?

**Comment by friedpenguin** [comment]
Tested. Firefox on Linux in Private works fine.

**Comment by NickM-27** [comment]
Perhaps this is just a Firefox bug then

**Comment by twistedlincoln** [comment]
> What error do you see in Firefox with this mode?

There is no error.  Just a plain blank screen.

> Perhaps this is just a Firefox on windows bug then

Sorry, I forgot to include that I'm using Linux. I was able to reproduce on a few different distros with different Firefox versions.  It's also an issue using Firefox on Android.

**Comment by NickM-27** [comment]
There should still be errors in the browser logs

**Comment by twistedlincoln** [comment]
After enabling the "Browser Console", I uncovered the following error when in private browsing mode that is not present in normal mode:

"InvalidStateError: A mutation operation was attempted on a database that did not allow mutations."

**Comment by twistedlincoln** [comment]
A search of this error uncovers a clue. It looks to be a decade-old limitation with Firefox not supporting Indexed DB connections within private browsing mode: https://bugzilla.mozilla.org/show_bug.cgi?id=781982

I just tested a workaround:  setting dom.indexedDB.privateBrowsing.enabled to true via about:config -- this allows Frigate to load property.  Unfortunately this defeats a good amount of the purpose of private browsing mode as this causes data to be stored in local storage rather than memory.  It looks like this was fixed in Firefox 115.  

So perhaps a reasonable way to address this it to provide an error message prompting the user to either update their browser or turn off private browsing?  

Although I'm guessing not many people have run into this issue as I did extensive searching prior to posting this bug and didn't come up with any other complaints.  So perhaps it may just resolve itself as time goes on and people update naturally.
